vermouth/pdb/pdb.py

```diff
@@ -443,6 +459,58 @@ def _do_single_conect(self, conect_record, id2idxs):
                             mol2.nodes[atomidx]['position'])
             mol.add_edge(atomidx0, atomidx, distance=dist)
 
+    def do_seqres(self):
+        """
+        Get the AA sequences of the chains in the molecules
+        """
+        chains = list(list(np.unique(list(mol.nodes[idx]['chain'] for idx in mol)))
+                      for mol in self.molecules)
+
+        properties = {a: [] for a in [x for xs in chains for x in xs]}
+
+        for line in self._seqres:
+            chain = line[11]
+            resnames = line[19:].split()
+            #some varient of this can check we have protein seqres and not
+            #seqres for nucleotides
+            #not much use atm but could be helpful for when we have nucleotides here
+            # try:
+            #     assert list(set(len(i) for i in resnames)) == 3
+            # except AssertionError:
+            #     #we don't really need to worry about this
+            #     #we just need to not use it any further
+            #     pass
+            properties[chain].append(resnames)
+
+        # this to make a flat list for each chain in the file
+        for chain in properties.keys():
+            properties[chain] = [x for xs in properties[chain] for x in xs]
+        self._check_seqres(properties)
+
+
+    def _check_seqres(self, properties):
+        '''
+        In future this could just be done by parsing REMARK 465.
+        '''
+        LOGGER.info("Checking pdb SEQRES entry for missing residues", type="step")
+        for mol in self.molecules:
+            resids = np.array([mol.nodes[idx]['resid'] for idx in mol],dtype = int)
+            chain = list(set([mol.nodes[idx]['chain'] for idx in mol]))[0]
+
+            missing_res_nos = np.setdiff1d(np.arange(len(properties[chain]))+1,
+                                       np.unique(resids)[np.unique(resids) < len(properties[chain])+1])
```

Comment:
Needs a comment :)
Does this assume resid numbers are sane?

Reply:
added. The < in the pdb residues array ensures that any residues that are larger (eg. for solvent molecules attached to the chain) are excluded, if that's what you mean by sanity?

Reply:
Does this mean that the residnumbers have to be in order? Do solvents etc *always* have to be last? I guess the real solution is to do the alignment?

Reply:
I don't think the resids have to necessarily be in order, they just have to exist. I agree that it's a problem if there are solvent residues listed first though, not sure how to get around that... 

I think increasingly there may have to be some kind of alignment included. I've now found a problem if there are eg. expression tags, which will be listed in the seqres, but not actually be part of the 'proper' sequence. For this, either we bring back some kind of alignment, or I parse SEQADV for this purpose. I think the latter should be easy enough to do given it will only be used internally for SEQRES comparison. Thoughts?

Reply:
Your call. I don't enough about the pdb format (and not really time to look it up) to say anything about the SEQADV. I do think the alignment is (going to be) important, but that can be medium-long term. For now I see 2 options: 1) SEQADV; 2) if it doesn't match, complain. You have to see whether that would give too many false positives, in which case you may want to make it an INFO message rather than warning.

Reply:
Why/when would alignment be important? In files that have sequence, it should correspond in a simple, literal way, so can be done without fancy alignment. But those files will also have REMARK 465, i dicating which residues in the sequence are not resolved. For files without sequence, there is nothing to align to, fancy or not.

Reply:
Iff the seqres does not match the found residues, where is it wrong? And can you guarantee that all pdb files that have a seqres have remark 465? I don't dare make *any* assumptions about pdb files, based on my experience so far.

Reply:
If the seqres does not match, someone has deliberately changed the coordinates section, and did not update SEQRES. I guess a deliberate change is meant to be processed. If there is a SEQRES, it's probably good to use it for sanity check, but not using NW or other alignment, because breaks may be assigned wrongly. Use it only for simple verbatim checking, preferably in conjunction with REMARK 465, and if it doesn't match, it makes sense to issue an error; the user can then discard the SEQRES record if that's what was meant.

Reply:
The question is more about what you do when the seqres doesn't match. 
1) You warn: "Seqres is wrong, good luck!"
2) You assume Remark 465 is correct and info after checking: "Seqres is wrong, but the deviations are specified in R465"
3) You warn: "Seqres is wrong and this is not described in R465. Probably, residues X-Z are missing, residue ALA53 was changed to PHE. Sort out your mess."

As user, I would like warning 3 most. But I'm completely fine with just 1 or 2.

Partly this also tied in with how much we trust CHAIN identifiers, but I guess it would be sane to assume they're at least consistent within the pdb file (i.e. between seqres and the coordinate section)